test/MinimalDelegation.execute.t.sol

```diff
@@ -176,48 +177,57 @@ contract MinimalDelegationExecuteTest is TokenHandler, DelegationHandler, Execut
+
         uint256 nonceKey = 0;
         (uint256 nonce,) = _buildNextValidNonce(nonceKey);
-        SignedCalls memory signedCalls = SignedCalls({calls: calls, nonce: nonce});
 
-        // Sign using the registered P256 key
-        bytes memory packedSignature = abi.encode(
-            nonce, abi.encode(p256Key.toKeyHash(), p256Key.sign(signerAccount.hashTypedData(signedCalls.hash())))
-        );
-        bytes memory executionData = abi.encode(calls, packedSignature);
+        bytes memory executionData = abi.encode(calls, _signAndPack(hash(calls, DEFAULT_NONCE), signerTestKey, nonce, signerTestKey.toKeyHash()));
 
+        vm.expectRevert(IKeyManagement.KeyDoesNotExist.selector);
         signerAccount.execute(BATCHED_CALL_SUPPORTS_OPDATA, executionData);
-        assertEq(signerAccount.getKey(secp256k1Key.toKeyHash()).hash(), secp256k1Key.toKeyHash());
     }
 
-    function test_execute_batch_opData_twoCalls_succeeds() public {
-        Call[] memory calls = CallBuilder.init();
+    // Root EOA must use bytes32(0) as their keyHash
+    function test_execute_batch_opData_rootEOA_withKeyHashZero_succeeds() public {
+        Call[] memory calls = CallUtils.init();
+        calls = calls.push(buildTransferCall(address(tokenA), address(receiver), 1e18)); // Transfer 1 tokenA
+
+        uint256 nonceKey = 0;
+        (uint256 nonce,) = _buildNextValidNonce(nonceKey);
+
+        bytes memory executionData = abi.encode(calls, _signAndPack(hash(calls, DEFAULT_NONCE), signerTestKey, nonce, ROOT_KEY_HASH));
+
+        signerAccount.execute(BATCHED_CALL_SUPPORTS_OPDATA, executionData);
+        assertEq(tokenA.balanceOf(address(receiver)), 1e18);
+    }
+
+    function test_execute_batch_opData_rootEOA_twoCalls_succeeds() public {
+        Call[] memory calls = CallUtils.init();
         calls = calls.push(buildTransferCall(address(tokenA), address(receiver), 1e18)); // Transfer 1 tokenA
         calls = calls.push(buildTransferCall(address(tokenB), address(receiver), 1e18)); // Transfer 1 tokenB
 
         // Get the current nonce components for key 0
         uint256 nonceKey = 0;
         (uint256 nonce, uint64 seq) = _buildNextValidNonce(nonceKey);
 
-        // Create hash of the calls + nonce and sign it
-        SignedCalls memory signedCalls = SignedCalls({calls: calls, nonce: nonce});
-        bytes32 hashToSign = signerAccount.hashTypedData(signedCalls.hash());
-        (uint8 v, bytes32 r, bytes32 s) = vm.sign(signerPrivateKey, hashToSign);
-        bytes memory signature = abi.encodePacked(r, s, v);
-
-        // Pack the execution data:
-        // 1. Encode the nonce and signature into opData
-        bytes memory opData = abi.encode(nonce, signature);
-        // 2. Encode the calls and opData together
-        bytes memory executionData = abi.encode(calls, opData);
+        bytes memory executionData = abi.encode(calls, _signAndPack(hash(calls, DEFAULT_NONCE), signerTestKey, nonce, ROOT_KEY_HASH));
```

Comment:
strange that this changed the gas costs for snapshotGasLastCall...

Reply:
should be fixed now, I was adding bytes32(0) to the root key when signing which wasn't what we were doing on main